vortex-array/src/array/primitive/stats.rs

```diff
@@ -189,26 +190,27 @@ impl<T: PStatsType> StatsAccumulator<T> {
             self.nan_count += 1;
         }
 
-        if self.prev == next {
+        let prev_cmp = next.total_cmp(self.prev);
+
+        if matches!(prev_cmp, Ordering::Equal) {
             self.is_strict_sorted = false;
         } else {
-            if next < self.prev {
+            if matches!(prev_cmp, Ordering::Less) {
                 self.is_sorted = false;
             }
             self.run_count += 1;
         }
-        if next < self.min {
+        if matches!(next.total_cmp(self.min), Ordering::Less) {
             self.min = next;
-        } else if next > self.max {
+        } else if matches!(next.total_cmp(self.max), Ordering::Greater) {
             self.max = next;
         }
         self.prev = next;
     }
 
     pub fn finish(self) -> StatsSet {
         let is_constant = (self.min == self.max && self.null_count == 0 && self.nan_count == 0)
-            || self.null_count == self.len
-            || self.nan_count == self.len;
```

Comment:
we should never claim nan's can be made constant for now. While the comparison now uses bit patters which would mean that equal nans could be really equal the logic here has to change from pure is_nan comparison